views/js/testRunner/testMetaData.js

```diff
@@ -0,0 +1,185 @@
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+/**
+ * Metadata to be sent to the server. Will be saved in result storage as a trace variable.
+ * Usage example:
+ * <pre>
+ * var testMetaData = new TestMetaData({
+ *   testServiceCallId : this.itemServiceApi.serviceCallId
+ * });
+ *
+ * testMetaData.setData({
+ *   'TEST' : {
+ *      'TEST_EXIT_CODE' : 'T'
+ *   },
+ *   'SECTION' : {
+ *      'SECTION_EXIT_CODE' : 704
+ *   }
+ * });
+ *
+ * testMetaData.addData({'ITEM' : {
+ *      'ITEM_START_TIME_CLIENT' : 1443596730143,
+ *      'ITEM_END_TIME_CLIENT' : 1443596731301
+ *    }
+ * });
+ * </pre>
+ *
+ * Data for each service call id will be stored in local storage to be able get data
+ * after reloading the page or resuming the test session.
+ *
+ * To clear all data related to current test_call_id used <i>clearData</i> method.
+ */
+define([
+    'lodash'
+], function (_) {
+    'use strict';
+
+    /**
+     *
+     * @param {Object} options
+     * @param {string} options.testServiceCallId - test call id.
+     * @constructor
+     */
+    function TestMetaData(options) {
+        var self = this,
+            _testServiceCallId,
+            _storageKeyPrefix = 'testMetaData_',
+            _data = {};
+
+        if (!options || options.testServiceCallId === undefined) {
+            throw new TypeError("testServiceCallId option is required");
+        }
+
+        this.SECTION_EXIT_CODE = {
+            'COMPLETED_NORMALLY': 700,
+            'QUIT': 701,
+            'COMPLETE_TIMEOUT': 703,
+            'TIMEOUT': 704,
+            'FORCE_QUIT': 705,
+            'IN_PROGRESS': 706,
+            'ERROR': 300
+        };
+        this.TEST_EXIT_CODE = {
+            'COMPLETE': 'C',
+            'TERMINATED': 'T',
+            'INCOMPLETE': 'IC',
+            'INCOMPLETE_QUIT': 'IQ',
+            'INACTIVE': 'IA',
+            'CANDIDATE_DISAGREED_WITH_NDA': 'DA'
+        };
+
+        /**
+         * Initialize test meta data manager
+         */
+        function init() {
+            _testServiceCallId = options.testServiceCallId;
+            self.setData(getLocalStorageData());
+        }
+
+        /**
+         * Set data to local storage
+         * @param {* }val - data to be stored.
+         */
+        function setLocalStorageData(val) {
+            localStorage.setItem(self.getLocalStorageKey(), val);
```

Comment:
Why do you need localStorage ? 
If you really need it, use `window.localStorage` and add a function to clean it up (and it must be cleaned-up at some point - remember the size is limited per origin).